helpers/sortAndFilterHotels.js

```diff
@@ -17,27 +16,48 @@ module.exports = function(dust) {
    * @example {@_sortAndFilterHotels parent=packageRatesReply.packageRates node="grossPrice" filterArray=_brandConfig.featuredHotel filterValue="x.links.hotelProducts.ids[0]" include="true"} {/_sortAndFilterHotels} output loop of parent object sorted by node and filtered by filterArray
    */
 
-  dust.helpers._sortAndFilterHotels = function(chunk, context, bodies, params) {
+  dust.helpers._sortAndFilterHotels = function( chunk, context, bodies, params ) {
     var filterArray = params.filterArray;
 
-    var mappedObject = sortBy(params.parent, params.node);
+    var mappedObject = sortBy( params.parent, params.node );
     var newMappedObject = [];
 
+    // check whether hotels in the filterArray are in the overall list
+    var mappedObjectContainsFilterArray =
+      mappedObject
+      .map( function( object ) {
+        // get the IDs of all hotels in the list
+        return object.links.hotelProducts.ids[ 0 ];
+      } )
+      .filter( function( hotelID ) {
+        // only return hotels with the ID of those in the filterArray
+        return filterArray.indexOf( hotelID ) > -1;
+      } )
+      //count
+      .length;
```

Comment:
Where I'm using it, I wanted it to be explicit that I"m checking for the existence of the hotels in the list